### PR TITLE
Automate AWS CDK and Terraform tests w/ GitHub Actions

### PR DESCRIPTION
**Pull request by robertlcx**

### CODE REVIEW

**Review by cabeaulac — COMMENTED** [review]
one little change

**Review by cabeaulac — APPROVED** [review]
Yay!

File: README.md [hunk 1]
```diff
@@ -68,18 +82,18 @@ From the working directory:
    your actual LocalStack key.
 
       ```bash
-      export LOCALSTACK_API_KEY=<your key>
+      export LOCALSTACK_AUTH_TOKEN=<your key>
```
Comment (cabeaulac) [inline README.md:85]: Maybe change <your key> to <your auth token> so it's not confused with the deprecated api keys

File: .github/workflows/weekly_test.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        . .venv/bin/activate
+        make setup-aws
+
+        # Unset stale org secret
+        printenv
+        unset LOCALSTACK_API_KEY
+
+        # Use repo secret and start Localstack in background
+        echo "Starting Localstack"
+        LOCALSTACK_AUTH_TOKEN="${{ secrets.LOCALSTACK_AUTH_TOKEN }}" DOCKER_COMPOSE_FLAGS="-d --timeout=30" make start-localstack
```
Comment (cabeaulac) [inline .github/workflows/weekly_test.yml:70]: This starts LS with the docker-compose that maps ports to the host. We don't want to map anything to the host in a CI pipeline. We want LS and a dev container on the same docker network, completely isolated. And, we have to pass in a unique pipeline run key to use in the compose name so concurrent runs work. Maybe we should do all that in the next PR and just get this one baselined first?

**Review by cabeaulac — CHANGES_REQUESTED** [review]
One simple update to comment in README. Other more expansive changes to running concurrent CI pipelines we can do in another PR I think.

File: README.md (hunk 1 above)
Comment (robertlcx) [reply README.md:85]: Change to what?

File: .github/workflows/weekly_test.yml (hunk 2 above)
Comment (robertlcx) [reply .github/workflows/weekly_test.yml]: This is a good observation. On it.

File: iac/terraform/cdk/appStack.ts [hunk 3]
```diff
@@ -40,13 +40,13 @@ export class AppStack extends TerraformStack {
         super(scope, id)
         this.config = config
         console.log('config', config)
-
-        let arch = 'arm64'
-        const localArch = process.env.LOCAL_ARCH
-
-        if (config.isLocal && localArch == 'x86_64') {
-            arch = 'x86_64'
+
```
Comment (cabeaulac) [inline iac/terraform/cdk/appStack.ts:43]: This is defaulting the architecture to undefined which is using the default `amd64` on my Mac M2 which isn't the intent. 
This is the code to run on `arm64` on my M2 and set to default to arm64 in AWS.
```
        let archList = ['arm64']
        const localArch = process.env.LOCAL_ARCH

        if (config.isLocal && localArch == 'x86_64') {
            archList = ['x86_64']
        }
```
Pic with your code 

Screenshot 2023-12-20 at 11 11 45 AM

Pic with code to set to native architecture on laptop and default to arm64 on AWS.
Screenshot 2023-12-20 at 11 17 34 AM

**Review by cabeaulac — CHANGES_REQUESTED** [review]
Architecture is not defaulting to arm64 on my Mac M2.

File: iac/terraform/cdk/appStack.ts (hunk 3 above)
Comment (robertlcx) [reply iac/terraform/cdk/appStack.ts:43]: Resolved now.

File: README.md (hunk 1 above)
Comment (robertlcx) [reply README.md:85]: @cabeaulac bumping this up.

**Review by cabeaulac — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (cabeaulac) [reply README.md:85]: I mean change 
```
export LOCALSTACK_AUTH_TOKEN=<your key>
```
to
```
export LOCALSTACK_AUTH_TOKEN=<your auth token>
```

sorry for the confusion

**Pull request merged**